### PR TITLE
Add Autoregressive distribution

### PR DESCRIPTION
**Pull request by Sayam753**
Solving #198 
This PR is not ready to be merged. Changes - 
 - Added AR distribution in timeseries.py
 - Added timeseries in distributions/__init__.py

I am not sure if [initial_state_prior](https://github.com/tensorflow/probability/blob/v0.9.0/tensorflow_probability/python/sts/autoregressive.py#L306:L311) can work with pymc4 distributions.

### CODE REVIEW

**Comment by lucianopaz** [comment]
> I am not sure if initial_state_prior can work with pymc4 distributions.

Not exactly. We could get it to work but I think that we should not have an `initial_state_prior` as a parameter that takes a `Distribution`. We should have an `initial_state` parameter that accepts a `Tensor` as input.

**Comment by Sayam753** [comment]
> > I am not sure if initial_state_prior can work with pymc4 distributions.
> 
> Not exactly. We could get it to work but I think that we should not have an `initial_state_prior` as a parameter that takes a `Distribution`. We should have an `initial_state` parameter that accepts a `Tensor` as input.

I will add that. 
Also should we add another parameter `order` in `AR` distribution or is it fine to calculate `order` as [done here](https://github.com/tensorflow/probability/blob/v0.9.0/tensorflow_probability/python/sts/autoregressive.py#L182).

**Comment by lucianopaz** [comment]
Let's not use an order parameter.

About the failed build, you're having lint problems. To fix them, from the project's root directory you have to run `make black`. If you are running windows, you have to run `python -m black pymc4 tests` instead

**Comment by Sayam753** [comment]
The `sample` method seems to work correctly
```python
>>> ar_model = AR('my-regressor', num_timesteps=5, coefficients=[-0.2, 0.8], level_scale=1)
>>> ar_model.sample()
<tf.Tensor: shape=(5, 1), dtype=float32, numpy=
array([[ 0.5804499 ],
       [ 0.02004898],
       [-0.62362194],
       [ 0.70075184],
       [ 1.396882  ]], dtype=float32)>
>>> 
>>> pm.evaluate_model(ar_model)
(<tf.Tensor: shape=(5, 1), dtype=float32, numpy=
array([[ 0.5592842 ],
       [-0.6405227 ],
       [ 0.5006866 ],
       [-0.41925687],
       [ 0.03968433]], dtype=float32)>, SamplingState(
    untransformed_values: ['my-regressor']
    transformed_values: []
    observed_values: []
    distributions: ['AR:my-regressor']
    num_potentials=0
    deterministics: []
    posterior_predictives: []))
```

File: pymc4/distributions/timeseries.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    """
+
+    def __init__(
+        self,
+        name,
+        num_timesteps,
+        coefficients,
+        level_scale,
+        initial_state=None,
+        observed_time_series=None,
```
[the remark this answers is not in the document]
Comment (Sayam753) [reply pymc4/distributions/timeseries.py]: Removed `observed_time_series` in recent commit

File: pymc4/distributions/timeseries.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            observed_time_series = tf.convert_to_tensor(value=observed_time_series)
+            observed_time_series = tf.concat([initial_state, observed_time_series], axis=0)
+
+        time_series_object = sts.Autoregressive(
+            order=order, observed_time_series=observed_time_series
+        )
+        distribution = time_series_object.make_state_space_model(
+            num_timesteps=num_timesteps,
+            param_vals={"coefficients": coefficients, "level_scale": level_scale},
+            initial_state_prior=time_series_object.initial_state_prior,
```
[the remark this answers is not in the document]
Comment (Sayam753) [reply pymc4/distributions/timeseries.py]: Even if `initial_state_prior` is not specified in `make_state_space_model`, sts.Autoregressive will create a default [tfd.MultivariateNormalDiag](https://github.com/tensorflow/probability/blob/v0.9.0/tensorflow_probability/python/sts/autoregressive.py#L369:L373) instance of `order` variate Gaussian.

[the remark this answers is not in the document]
Comment (lucianopaz) [reply pymc4/distributions/timeseries.py]: Then we have to bypass the default prior. We'll have to use `tfd.Deterministic` and pass it as the `initial_state_prior` of the `make_state_space_model`. That way, we'll give it something that works as if it were a distribution but is in fact a tensor

[the remark this answers is not in the document]
Comment (Sayam753) [reply pymc4/distributions/timeseries.py]: The base distribution of `make_state_space_model` (LinearGaussianStateSpaceModel) accepts `initial_state_prior` as an instance of [MultivariateNormalLinearOperator](https://github.com/tensorflow/probability/blob/356cfddef026b3339b8f2a81e600acd2ff8e22b4/tensorflow_probability/python/distributions/linear_gaussian_ssm.py#L331:L333). To achieve Deterministic nature and bypass default prior, is it fine to use `tfd.MultivariateNormalDiag` with mean as `initial_state` values and stddev close to 0 ?

[the remark this answers is not in the document]
Comment (lucianopaz) [reply pymc4/distributions/timeseries.py]: That's quite dissatisfying. Let's just focus for now on writing some small tests for forward sampling and log_prob evaluations. Later we'll worry about this problem

[the remark this answers is not in the document]
Comment (Sayam753) [reply pymc4/distributions/timeseries.py]: Added tests for forward sampling and log_prob evaluations in b4f6d80.

**Pull request merged**